### PR TITLE
Updates the Grid CLI to send Sabre transactions

### PR DESCRIPTION
**Pull request by agunde406**
This is built on top of #50 

This includes the version of the sabre-sdk to 0.3 and updating the grid cli to send all transactions wrapped in a sabre transaction.

You can test this by running docker-compose
`docker-compose up --build`

move to the grid/cli and create a new pike organization with the following command:
`cargo run -- -vv --url http://localhost:8080 organization create org_1 OrgTest test_address`

You will see the sabre tp execute a transaction and then you can see the organization by checking hte restapi:
`http://localhost:8080/organization`

### CODE REVIEW

File: sdk/src/protocol/sabre/payload.rs [hunk 1]
```diff
@@ -0,0 +1,3422 @@
...
+#[derive(Default, Clone)]
+pub struct CreateSmartPermissionActionBuilder {
+    name: Option<String>,
+    org_id: Option<String>,
+    function: Vec<u8>,
+}
+
+impl CreateSmartPermissionActionBuilder {
+    pub fn new() -> Self {
+        CreateSmartPermissionActionBuilder::default
```
Comment (vaporos) [inline sdk/src/protocol/sabre/payload.rs]: This and maybe some of the other structs would more naturally be an enum. (Take a look at the TnT protocol code, I think there is an example there.)

**Comment by vaporos** [comment]
Should the protocol piece go into sabre, with this code pulling in the sabre SDK?

File: sdk/src/protocol/sabre/payload.rs (hunk 1 above)
Comment (agunde406) [reply sdk/src/protocol/sabre/payload.rs]: Will take a look

**Comment by agunde406** [comment]
The Sabre SDK currently is only used for writing Sabre Smart Contracts. If it was added to the sabre sdk, it should be expanded to include the state objects as well and update the rest of the sabre code to use the protocols (instead of protos) Is that what you are suggesting?

**Comment by vaporos** [comment]
Yes, I think that is the logical conclusion of what I'm proposing.

File: cli/src/transaction.rs [hunk 2]
```diff
@@ -150,3 +213,61 @@ fn bytes_to_hex_str(b: &[u8]) -> String {
...
+    String::from(SABRE_CONTRACT_PREFIX) + &bytes_to_hex_str(hash)[..64]
+}
+
+/// Returns a state address for a given namespace registry
+///
+/// # Arguments
+///
+/// * `namespace` - the address prefix for this namespace
+fn compute_namespace_registry_address(namespace: &str) -> Result<String, CliError> {
+    let prefix = match namespace.get(..6) {
```
Comment (peterschwarz) [inline cli/src/transaction.rs]: Not necessary to change, but you could do this as:
```
let prefix = namespace.get(..6).ok_or_else(||
    Err(CliError::UserError(format!(
        "Namespace must be at least 6 characters long: {}",
        namespace
    ))))?;
```

**Pull request merged**